### PR TITLE
feat(hooks): implement useClearRefinements

### PR DESCRIPTION
**Pull request by Haroenv**
**Summary**

adds the clearRefinements hook, as well as using it in the example.

**Result**

FX-737
#3252

### CODE REVIEW

File: examples/hooks/components/ClearRefinements.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  return (
+    <div className={cx('ais-ClearRefinements', props.className)}>
+      <button
+        className={cx(
+          'ais-ClearRefinements-button',
+          !canRefine && 'ais-ClearRefinements-button--disabled'
+        )}
+        disabled={!canRefine}
+        onClick={refine}
```
Comment (francoischalifour) [inline examples/hooks/components/ClearRefinements.tsx:22]: This is a bit less misleading – wdyt?

```suggestion
        onClick={() ⇒ refine()}
```

File: packages/react-instantsearch-hooks/src/__tests__/useClearRefinements.test.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import { renderHook } from '@testing-library/react-hooks';
+import React from 'react';
+
+import { createInstantSearchTestWrapper } from '../../../../test/utils';
+import { useClearRefinements } from '../useClearRefinements';
+import { useRefinementList } from '../useRefinementList';
+
+import type { UseRefinementListProps } from '..';
```
Comment (francoischalifour) [inline packages/react-instantsearch-hooks/src/__tests__/useClearRefinements.test.tsx:8]: We can group it with this import line 6.

**Review by francoischalifour — APPROVED** [review]

File: packages/react-instantsearch-hooks/src/__tests__/useClearRefinements.test.tsx (hunk 2 above)
Comment (Haroenv) [reply packages/react-instantsearch-hooks/src/__tests__/useClearRefinements.test.tsx:8]: it was copied from another one, maybe use use eslint to be consistent later?

File: examples/hooks/components/ClearRefinements.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  return (
+    <div className={cx('ais-ClearRefinements', props.className)}>
+      <button
+        className={cx(
+          'ais-ClearRefinements-button',
+          !canRefine && 'ais-ClearRefinements-button--disabled'
+        )}
+        disabled={!canRefine}
+        onClick={() ⇒ refine()}
```
Comment (Haroenv) [inline examples/hooks/components/ClearRefinements.tsx:22]:
```suggestion
        onClick={() => refine()}
```

**Pull request merged**